Deprecated: Output an informational message for deprecation when no version provided

gziolo opened the pull request:
## Description

When there is no `options.version` param provided `deprecated` method outputs more relaxed warning message to the Web Console.

The fact that we warn when there is no clear path for removing deprecation is very confusing in my opinion. The change proposed tries to make it less dramatic for 3rd party developers. The idea is that when there is no version provided we are going to support deprecated version forever according to [Deprecations document](https://github.com/WordPress/gutenberg/blob/master/docs/designers-developers/developers/backward-compatibility/deprecations.md). As soon as we decide to remove it, we should specify the target version for removal and it will automatically make the message stronger. 

### Before

![Screen Shot 2019-07-26 at 18 27 32](https://user-images.githubusercontent.com/699132/61966503-0d714c80-afd3-11e9-85a5-8212ef009213.png)
![Screen Shot 2019-07-26 at 18 24 49](https://user-images.githubusercontent.com/699132/61966445-eb77ca00-afd2-11e9-853b-87ad6f17a2c4.png)


### After


![Screen Shot 2019-07-31 at 14 03 47](https://user-images.githubusercontent.com/699132/62210423-6c9cdb80-b39c-11e9-9b0c-97571c188697.png)
![Screen Shot 2019-07-31 at 14 05 42](https://user-images.githubusercontent.com/699132/62210424-6d357200-b39c-11e9-8e85-82bbe1489945.png)

gziolo committed:
Deprecated: Output an informational message for deprecation when no version provided
When there is no `options.version` param provided `deprecated` method outputs an informational message to the Web Console rather than warns as previously.

gziolo commented:
@aduth, should we add `info` to the list of observed console message types in e2e tests?
https://github.com/WordPress/gutenberg/blob/32b7b341885c6e5d9b710bfe4ed64123d3e23a9d/packages/e2e-tests/config/setup-test-framework.js#L32-L35

aduth commented:
Hm, I like the updated string, but I feel like we should always warn when something is deprecated. It shouldn't be interpreted as advisory, it should be a motivator to move away from.

draganescu commented:
I tend to agree with @aduth that deprecations should be a warning indifferent of version. It is one of the ways that pushes people to work on stuff they don't like, all these warnings, while informational messages are not motivating enough :) 

Something to add is the word "version" to this string: "%feature% is deprecated and will be removed in _version_ 11"

gziolo committed:
Use warn for all deprecations

gziolo commented:
@aduth and @draganescu - I changed back to `console.warn` to be used in all cases. I also added the `version` keyword in the message when appropriate.

aduth commented on packages/deprecated/src/test/index.js:
```diff
@@ -19,23 +19,23 @@ describe( 'deprecated', () => {
 
 		expect( console ).toHaveWarnedWith(
-			'Eating meat is deprecated and will be removed in the future.'
+			'Eating meat is deprecated and will be removed in version the future.'
```
This doesn't seem like something we'd want, nor what would be expected. Might be what's failing the build 😄

gziolo replied:
Build fails temporarily because of the known issue with annotations test.

I think this unit test is a bit misleading.

If you provide a number, it makes more sense:

`Eating meat is deprecated and will be removed in version 12.0.`

aduth replied:
Oh, I see. Maybe we should update the test cases to avoid it reading so awkwardly?

gziolo replied:
Yep, I hesitate to do it in the first place but I will do it now 👍

gziolo replied:
https://github.com/WordPress/gutenberg/pull/16774/commits/0a048a30cf999ce8859224163aa2c3a08c85f176 - addressed

gziolo committed:
Update unit tests for deprecated method to make them less confusing around version param

aduth commented:
Deprecated enhancements got me thinking: #16842

gziolo commented:
> Deprecated enhancements got me thinking: #16842

Yeah, very cool ideas 💯
